### PR TITLE
Adding option to disable pdf/epub builds for sphinx based projects.

### PR DESCRIPTION
**Pull request by gregmuellegger**
As propsed in #1154 we should have the ability to disable pdf and epub builds.

This pull request implements the requested feature.

### CODE REVIEW

**Comment by gregmuellegger** [comment]
I currently have the problem locally that the migration for adding the fields sets `sphinx_enable_pdf_build` and `sphinx_enable_epub_build` to `False` by default, which is stated differently in the migration.

It might be a problem with my setup using sqlite locally, but we should test this further before deploying since that would change the behaviour of all existing projects.

**Comment by ericholscher** [comment]
Hmm, yea those should definitely default to True. Perhaps we could set the model to default=True for now, generate the migration, then change it?

**Comment by ericholscher** [comment]
This looks great. :+1: -- Glad to see some more test coverage here. 

My only thought is that it might be good to expand this into it's own "build formats" concept that lives outside of the project. I don't know if people will want different formats for different versions, or anything like that. Perhaps that's overengineering. I do know that if we start adding more formats, having them be checklists on the Project Admin will feel weird.  

This is a great first step, and is an easy way to get us closer to where we need to be, so big +1.

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -114,6 +113,16 @@ class Project(models.Model):
         help_text=_("Google Analytics Tracking ID (ex. UA-22345342-1). "
                     "This may slow down your page loads."))
 
+    # Sphinx specific build options.
```
Comment (ericholscher) [inline readthedocs/projects/models.py]: I think we should probably leave the "sphinx" naming off these. They should be global settings for the project, and respected where they need to be, but having "sphinx" in the name feels weird.

Comment (ericholscher) [inline readthedocs/projects/models.py]: I'm happy to keep this PR as is, and address the naming stuff in a v2 where we move it into a completely separate model, or into a YAML file, as well, if it's a lot of work to rename all this stuff.

Comment (gregmuellegger) [inline readthedocs/projects/models.py]: No it's really straightforward to rename the field again. It should be a simple search&replace. I will do so on Thursday.

**Comment by gregmuellegger** [comment]
> I don't know if people will want different formats for different versions, or anything like that. Perhaps that's overengineering.

Yes I think that's a little complex for now especially for the interface in the admin. That will be a nobrainer once a readthedocs.yml file is in place. That file will always come from the version that the build is triggered from, which therefore allows to enable/disable pdf/epub builds per version.

**Comment by gregmuellegger** [comment]
I renamed the fields to be more generic in their name and not include the `sphinx_` prefix. Is it ready for merge then?

**Comment by ericholscher** [comment]
Yep, looks good to me. I will make sure the flags get set to True on our prod migrations.

**Comment by agjohnson** [comment]
We should probably have a fix for this migration, or a data migration that addresses this instead of hacking it in prod. I'll try for a solution today before I deploy.

**Comment by tisdall** [comment]
It seems like after shutting off PDF generation there still exists a link to the previously generated PDF and the previously generated PDF is still available.  Here's an example with PDF creation unchecked:  http://colanderalchemy.readthedocs.org

**Comment by stevepiercy** [comment]
The link is in the RTD popup menu
RTD popup menu.

**Comment by orsenthil** [comment]
Thank you!

**Pull request merged**